Userland/Libraries/LibGfx/Path.cpp

```diff
@@ -393,136 +393,227 @@ struct RoundTrip {
     ReadonlySpan<T> m_span;
 };
 
-Path Path::stroke_to_fill(float thickness) const
+Vector<Tuple<float, float>> Path::compute_dash_positions(float path_length, StrokeProperties const& stroke_properties) const
 {
-    // Note: This convolves a polygon with the path using the algorithm described
-    // in https://keithp.com/~keithp/talks/cairo2003.pdf (3.1 Stroking Splines via Convolution)
+    // https://www.w3.org/TR/svg-strokes/#StrokeShape
+
+    // FIXME: what about dashadjust???
+    // FIXME: what about dashcorner???
+
+    // Let pathlength be the length of the subpath.
+    float pathlength = path_length;
+    // Let dashes be the list of values of ‘stroke-dasharray’ on the element, converted to user
+    // units, repeated if necessary so that it has an even number of elements; if the property
+    // has the value none, then the list has a single value 0.
+    // FIXME: Convert to user units
+    Vector<float> dashes = stroke_properties.stroke_dasharray.size() > 0 ? stroke_properties.stroke_dasharray : (Vector<float>) { 0 };
```

Comment:
I am not really sure, for the algorithm implemented right now there would be no problem (even when `stroke_dasharray` is empty because `sum` gets initialized to 0) but I think the way to implement `stroke-dashadjust` is to modify the `dashes` array. Maybe it is better to change it for now and deal with the adjust when it gets implemented...